### PR TITLE
More cleanup and support for concrete as part of a project hierarchy.

### PR DESCRIPTION
**Pull request by azenla**

### CODE REVIEW

**Comment by azenla** [comment]
I would recommend not releasing after this as I have a few more changes I want to make.

**Comment by azenla** [comment]
This should be ready to merge, it adds a library concept so that you can have libraries where the bukkit API is accessible but doesn't get counted as a plugin. Also added is a GitHub workflow for making sure the plugin builds.

File: samples/mixed/bukkit-plugins/common-library/src/main/kotlin/lgbt/mystic/foundation/conrete/sample/common/CommonExtensions.kt [hunk 1]
```diff
@@ -0,0 +1,6 @@
+package lgbt.mystic.foundation.conrete.sample.common
```
Comment (kubeliv) [inline samples/mixed/bukkit-plugins/common-library/src/main/kotlin/lgbt/mystic/foundation/conrete/sample/common/CommonExtensions.kt]: `conrete`

File: samples/simple/gradle/wrapper/gradle-wrapper.properties [hunk 2]
```diff
@@ -0,0 +1,5 @@
+distributionBase=GRADLE_USER_HOME
+distributionPath=wrapper/dists
+distributionUrl=https\://services.gradle.org/distributions/gradle-7.4-bin.zip
```
Comment (kubeliv) [inline samples/simple/gradle/wrapper/gradle-wrapper.properties]: Could this be bumped to 7.4.2 please?

File: src/main/kotlin/lgbt/mystic/foundation/concrete/extensions.kt [hunk 3]
```diff
@@ -1,16 +1,50 @@
 package lgbt.mystic.foundation.concrete
```
Comment (kubeliv) [inline src/main/kotlin/lgbt/mystic/foundation/concrete/extensions.kt:1]: Could you please add a bit of documentation to this class?

**Review by kubeliv — CHANGES_REQUESTED** [review]
There are some quick changes I'd like addressed, but the rest looks good. I'll rereview once they're made.

File: src/main/kotlin/lgbt/mystic/foundation/concrete/extensions.kt (hunk 3 above)
Comment (kubeliv) [reply src/main/kotlin/lgbt/mystic/foundation/concrete/extensions.kt:1]: Or file, rather.

**Comment by kubeliv** [comment]
Looks good, thanks.

**Pull request merged**